compiler/semobjconstr.nim

```diff
@@ -380,6 +380,8 @@ proc defaultConstructionError(c: PContext, t: PType, info: TLineInfo) =
     assert false, "Must not enter here."
 
 proc semObjConstr(c: PContext, n: PNode, flags: TExprFlags; expectedType: PType = nil): PNode =
+  if nfUseDefaultField in n.flags:
+    return n
```

Comment:
Why do we care? It's `semObjConstr`'s job to semcheck an object construction. It's not its job to "cache" the value or to detect if it already was semchecked. Why was it already semchecked? Where does that happen?

Reply:
```nim
import std/tables
block:
  type TokenData = object
    campaignMemberships = Table[string, string]()
 ```

`Table[string, string]()` is evaluated at the compile time. When it is passed to `semObjConstr`, it has been inlined to `Table(data: [], state: 0)` by `getConstExpr`. But `data` is a private field from another module. If it is checked here, the `semObjConst` will report `the field '$1' is not accessible.`. I think we should allow `semObjectConst` to ignore constant object construction.

Reply:
Sure, I will add `efSkipFieldVisibilityCheck` flag instead.

Reply:
Tried to semcheck the constant fields, but there is something wrong with the construction or type:

```
Table[string, string]()
------semConstExpr-------
Table(data: [], counter: 0)
```

```
tobject_default_value.nim(437, 13) template/generic instantiation of `main` from here
tobject_default_value.nim(424, 50) Error: type mismatch: got <array[0..-1, empty]> but expected 'KeyValuePairSeq[system.string, system.string]'
```
`data` is a seq type, but VM returns an empty array. It cannot pass `semconstr`.

Reply:
Now I add efSkipFieldVisibilityCheck and fixed the type of nkBracket.